src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Utils/GlobalLicenseDictionary.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+// I know use of a global state is not ideal. I could use feedback on how to achieve this in a better way.
+public static class GlobalLicenseDictionary
+{
+    private static Dictionary<string, string> _licenseDictionary = new Dictionary<string, string>();
```

Comment:
My suggestion above would have you move away from the static Dictionay to the SingletonClass, but I think you'll want to both move away from the Dictionary as the cache since this would grow unbounded if the parser were up for a long time. I also don't have a ton of domain context here, but I would assume that "PackageName" is not a complete key since the license might have changed from one version to the next?

Reply:
I like this idea of using a singleton and you're right the key should be "packageName" + "packageVersion". I'll add this.